### ISSUE TITLE
Money.Sigil requires existence of the atom for the currency

### ISSUE THREAD

**Issue by marcelotto** [kipcole9/money#46]
Since the implementation of `Money.Sigil` relies on `String.to_existing_atom/1` it can't be used without former usage of the atom for the currency:

```
Interactive Elixir (1.5.1) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)> import Money.Sigil
Money.Sigil
iex(2)> ~M[42]EUR
** (ArgumentError) argument error
    :erlang.binary_to_existing_atom("EUR", :utf8)
    (ex_money) lib/money/sigil.ex:34: Money.Sigil.atomize/1
    (ex_money) lib/money/sigil.ex:21: Money.Sigil.sigil_M/2
```

I've noticed [this similar issue](#8), but the fix got removed later.

**Comment by kipcole9** [comment]
Thanks for the bug report. Fixed in master and also published as [version 2.0.2 on hex](https://hex.pm/packages/ex_money).

**Comment by kipcole9** [comment]
I also researched [issue #8](https://github.com/kipcole9/money/issues/8) that you referenced.  I reintroduced this issue when I repackaged `ex_cldr` into several smaller dependencies which is not great. 

I haven't found a good way to test for this specific error since I don't know how to ensure a test has a "no modules loaded yet" state to start with.  I've added some tests for the negative case (i.e. raises error on invalid currency code in the sigil).

I'll ask the forums help for a way to unload all modules before a test runs - if you have any thoughts they'd be appreciated.
